### PR TITLE
Improve field alignment of structs

### PR DESCRIPTION
**Pull request by bshephar**
This change improves the field alignment in our structs to reduce the overall memory footprint required for each struct. We do this by aligning like types within the structs to reduce the amount of padding required to align each object in memory.

It's a simple re-ordering of objects with no functional change to how the operator works from the user perspective. See the explanation provided here regarding maligned structs:
https://medium.com/@sebassegros/golang-dealing-with-maligned-structs-9b77bacf4b97

### CODE REVIEW

**Comment by bshephar** [comment]
Hard to parse that failure, but it looks like a functional test failed:
```
43mwhen A dataplaneDeployment is created with two NodeSets \u001b[38;5;9m\u001b[1m[It] should have conditions set\u
```

**Comment by bshephar** [comment]
recheck

**Comment by bshephar** [comment]
recheck

**Comment by bshephar** [comment]
Seems like it failed to run functional tests on the content-provider job. But this seems to be passing locally:
```
Will run 61 of 61 specs
Running in parallel across 4 processes
•••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••

Ran 61 of 61 Specs in 22.719 seconds
SUCCESS! -- 61 Passed | 0 Failed | 0 Pending | 0 Skipped
```

Re-running to see if it was something intermittent.

**Comment by bshephar** [comment]
```
  The function passed to Eventually failed at /home/zuul/go/pkg/mod/github.com/openstack-k8s-operators/lib-common/modules/common@v0.3.1-0.20240306153230-dc65ab49ebc0/test/helpers/conditions.go:49 with:
  Ready condition is in an unexpected state. Expected: True, Actual: False, instance name: 8fbc7b61-faed-4c86-a8b8-0944f1d8a318/edpm-compute-nodeset-global, Conditions: [{Ready False Error 2024-03-13 21:48:05 -0400 EDT Error DataPlaneNodeSet error occurred nodeSet: beta-nodeset error: openstackansibleees.ansibleee.openstack.org \"foo-service-edpm-compute-nodeset-global\" already exists} {DeploymentReady True  2024-03-13 21:48:06 -0400 EDT Ready Deployment completed} {InputReady True  2024-03-13 21:48:05 -0400 EDT Ready Setup complete} {NodeSetDeploymentReady False Error 2024-03-13 21:48:05 -0400 EDT Error DataPlaneNodeSet error occurred nodeSet: beta-nodeset error: openstackansibleees.ansibleee.openstack.org \"foo-service-edpm-compute-nodeset-global\" already exists}]
  Expected
      <v1.ConditionStatus>: False
  to equal
      <v1.ConditionStatus>: True\u001b[0m
  \u001b[38;5;9mIn \u001b[1m[It]\u001b[0m\u001b[38;5;9m at: \u001b[1m/home/zuul/go/pkg/mod/github.com/openstack-k8s-operators/lib-common/modules/common@v0.3.1-0.20240306153230-dc65ab49ebc0/test/helpers/conditions.go:53\u001b[0m \u001b[38;5;243m@ 03/13/24 21:48:46.305\u001b[0m

  \u001b[38;5;9mFull Stack Trace\u001b[0m
    github.com/openstack-k8s-operators/lib-common/modules/common/test/helpers.(*TestHelper).ExpectCondition(0xc0002efc70, {{0xc00001ba40, 0x24}, {0x1ea226b, 0x1b}}, {0x214cac0?, 0x1f976d8}, {0x1e7e5bb, 0x5}, {0x1e7d992, ...})
    \t/home/zuul/go/pkg/mod/github.com/openstack-k8s-operators/lib-common/modules/common@v0.3.1-0.20240306153230-dc65ab49ebc0/test/helpers/conditions.go:53 +0x3bb
    github.com/openstack-k8s-operators/dataplane-operator/tests/functional.glob..func1.3.3()
    \t/home/zuul/src/github.com/openstack-k8s-operators/dataplane-operator/tests/functional/openstackdataplanedeployment_controller_test.go:421 +0x9e5
\u001b[38;5;243m------------------------------\u001b[0m

\u001b[38;5;9m\u001b[1mSummarizing 1 Failure:\u001b[0m
  \u001b[38;5;9m[FAIL]\u001b[0m \u001b[0mDataplane Deployment Test \u001b[38;5;243mwhen A dataplaneDeployment is created with two NodeSets \u001b[38;5;9m\u001b[1m[It] should have conditions set\u001b[0m
  \u001b[38;5;243m/home/zuul/go/pkg/mod/github.com/openstack-k8s-operators/lib-common/modules/common@v0.3.1-0.20240306153230-dc65ab49ebc0/test/helpers/conditions.go:53\u001b[0m

\u001b[38;5;9m\u001b[1mRan 61 of 61 Specs in 67.159 seconds\u001b[0m
\u001b[38;5;9m\u001b[1mFAIL!\u001b[0m -- \u001b[38;5;10m\u001b[1m60 Passed\u001b[0m | \u001b[38;5;9m\u001b[1m1 Failed\u001b[0m | \u001b[38;5;11m\u001b[1m0 Pending\u001b[0m | \u001b[38;5;14m\u001b[1m0 Skipped\u001b[0m

coverage: 57.4% of statements
composite coverage: 57.4% of statements

Ginkgo ran 1 suite in 1m18.125375332s
```


Will need to try reproducing it somewhere.

**Comment by slagle** [comment]
@bshephar do you plan to update this one?

**Comment by bshephar** [comment]
> @bshephar do you plan to update this one?

Updated

**Review by fao89 — COMMENTED** [review]
+2

**Review by slagle — APPROVED** [review]
